src/method/method_invoke_group.cpp

```diff
@@ -198,8 +206,15 @@ namespace cubmethod
 	  }
 
 	error = m_method_vector[i]->get_return (m_thread_p, arg_base, m_result_vector[i]);
+	if (m_rctx->is_interrupted ())
+	  {
+	    error = m_rctx->get_interrupt_reason ();
```

Comment:
Here, the return value of m_rctx->get_interrupt_reason() cannot be NO_ERROR. Also, you don't have to call m_rctx->set_interrupt_by_reason(error) again.
So, it looks like you could do something like this:
`return m_rctx->get_interrupt_reason();`

Reply:
At m_rctx->set_interrput (error), the `error` may not interrupt reason such as ER_INTERRUPTED. If an error is not interrupt-related code, It does not enter interrupt mode.